Meadowlark-js/packages/meadowlark-core/src/handler/NaturalKeyExtractor.ts

```diff
@@ -122,8 +123,8 @@ export function extractNaturalKey(entity: TopLevelEntity, body: object): Natural
  * TODO: Bring this behavior into extractNaturalKey, correctly managing for natural key component ordering.
  */
 export function deriveAssignableFrom(entity: TopLevelEntity, naturalKey: string): AssignableInfo | null {
-  const { assignableTo }: { assignableTo: TopLevelEntity | null } = (entity.data
-    .meadowlark as EntityMeadowlarkData).apiMapping;
+  const { assignableTo }: { assignableTo: TopLevelEntity | null } = (entity.data.meadowlark as EntityMeadowlarkData)
+    .apiMapping;
```

Comment:
I'm having trouble understanding what this `{ .. } : { ... } = ...` is actually doing. What is the data type on the left?

Reply:
This is the somewhat odd Typescript syntax for annotating an object destructuring assignment. entity.data.meadowlark.apiMapping is of type ApiEntityMapping.  ApiEntityMapping has a property assignableTo, and  assignableTo is of type TopLevelEntity | null.

Three options here: 1) Leave as is. 2) Drop it completely - not strictly necessary as TypeScript knows the type. 3) Introduce a local type alias like `type NullableTopLevelEntity = TopLevelEntity | null` and use that to replace `{ assignableTo: TopLevelEntity | null }`.

Reply:
Option 3 sounds good. This sort of thing is powerful I'm sure, but confusing for those of us who don't live and breathe TypeScript.